From d93c869e950c9e875992832d541eb47c6c0c2d53 Mon Sep 17 00:00:00 2001
From: James Hilliard <james.hilliard1@gmail.com>
Date: Sat, 10 Jul 2021 04:36:50 -0600
Subject: [PATCH 01/73] Disable non-explicit sbatvars autodetection for cross
 builds.

Since autodetection is unlikely to work reliably for cross builds
disable it unless explicitly enabled.

Signed-off-by: James Hilliard <james.hilliard1@gmail.com>
---
 src/boot/efi/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/boot/efi/meson.build b/src/boot/efi/meson.build
index afdf739d9..62d826bec 100644
--- a/src/boot/efi/meson.build
+++ b/src/boot/efi/meson.build
@@ -115,7 +115,7 @@ if have_gnu_efi
                             ['sbat-distro-url', 'BUG_REPORT_URL']]
                 foreach sbatvar : sbatvars
                         value = get_option(sbatvar[0])
-                        if value == '' or value == 'auto'
+                        if (value == '' and not meson.is_cross_build()) or value == 'auto'
                                 cmd = 'if [ -e /etc/os-release ]; then . /etc/os-release; else . /usr/lib/os-release; fi; echo $@0@'.format(sbatvar[1])
                                 value = run_command(sh, '-c', cmd).stdout().strip()
                                 message('@0@ (from @1@): @2@'.format(sbatvar[0], sbatvar[1], value))
-- 
2.34.1

